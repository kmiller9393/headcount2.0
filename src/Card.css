.Card {
  background-color: #f7f7f7;
  border-radius: 10px;
  box-shadow: inset 0 0 10px #000000;
  height: 25rem;
  margin: 0.5rem;
  -moz-box-shadow: inset 0 0 10px #000000;
  padding: 0 0.35rem 0 0.35rem;
  text-align: center;
  -webkit-box-shadow: inset 0 0 10px #000000;
  width: 15rem;
}

.Chosen {
  
}

h2 {
  font-size: 1.5rem;
  margin: 0.75rem 0 0.75rem 0;
}

ul {
  font-size: 0.8rem;
  margin: 0.25rem 0 0.25rem 0;
  padding: 0;
}

.low-data {
  color: coral;
}

.high-data {
  color: blue;
}
