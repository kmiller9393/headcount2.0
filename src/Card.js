import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './App.css';
import './Card.css';

export default class Card extends Component {
  constructor() {
    super();
    this.state = {
      selected: false
    };
  }

  handleClick = () => {
    const selectStatus = this.state.selected;
    this.setState({
      selected: !selectStatus
    });
    this.props.chooseCard(this.props.location);
  };

  render() {
    return (
      <div
        // className={this.state.selected ? 'Chosen' : 'Card'}
        className="Card"
        onClick={this.handleClick}
      >
        <h2>{this.props.location}</h2>
        {Object.keys(this.props.stats).map(stat => {
          return (
            <article>
              <ul
                className={
                  this.props.stats[stat] < 0.5 ? 'low-data' : 'high-data'
                }
              >
                {stat}: {this.props.stats[stat]}
              </ul>
            </article>
          );
        })}
      </div>
    );
  }
}

Card.propTypes = {
  stats: PropTypes.object,
  location: PropTypes.string
};
