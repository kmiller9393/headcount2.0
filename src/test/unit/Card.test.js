import React from 'react'
import Card from '../../Card'
import { shallow, mount } from 'enzyme'

describe('Card', () => {
  let wrapper
  let compareCardsMock
  let key
  let chooseCardMock
  let removeCardMock
  let compareDistricts = []

  beforeEach(() => {
    key = 1
    chooseCardMock = jest.fn()
    compareCardsMock = jest.fn()
    removeCardMock = jest.fn()
    wrapper = shallow(
      <Card
        location="COLORADO"
        stats={{ 2004: 0.24 }}
        key={key}
        selected
        compareDistricts={compareDistricts}
        chooseCard={chooseCardMock}
        compareCards={compareCardsMock}
        removeCard={removeCardMock}
      />
    )
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('the handleClick changes the state of the card when clicked', () => {
    wrapper = shallow(
      <Card
        location="COLORADO"
        stats={{ 2004: 0.24 }}
        key={key}
        compareDistricts={compareDistricts}
        chooseCard={chooseCardMock}
        compareCards={compareCardsMock}
        removeCard={removeCardMock}
      />
    )
    expect(wrapper.state('selected')).toEqual(false)
    wrapper.instance().handleClick()
    expect(wrapper.state('selected')).toEqual(true)
  })

  it('calls removeCard when it has a selected property and does not call chooseCard', () => {
    wrapper.instance().handleClick()
    expect(removeCardMock).toHaveBeenCalled()
    expect(chooseCardMock).not.toHaveBeenCalled()
  })

  it('should have a falsy property of selected and then chooseCard should be called', () => {
    wrapper = shallow(
      <Card
        location="COLORADO"
        stats={{ 2004: 0.24 }}
        key={key}
        compareDistricts={compareDistricts}
        chooseCard={chooseCardMock}
        compareCards={compareCardsMock}
        removeCard={removeCardMock}
      />
    )
    wrapper.instance().handleClick()
    expect(chooseCardMock).toHaveBeenCalled()
  })

  it('should have a class of Chosen when it has a compareDistricts property or a selected property', () => {
    expect(wrapper.find('div').hasClass('Chosen')).toEqual(true)
  })

  it('should not have a class of Chosen when it does not have a compareDistricts property or a selected property', () => {
    wrapper = shallow(
      <Card
        location="COLORADO"
        stats={{ 2004: 0.24 }}
        key={key}
        chooseCard={chooseCardMock}
        compareCards={compareCardsMock}
        removeCard={removeCardMock}
      />
    )
    expect(wrapper.find('div').hasClass('Chosen')).toEqual(false)
  })

  it('should have the low-data class when its stat is below 0.5', () => {
    wrapper = shallow(
      <Card
        location="COLORADO"
        stats={{ 2004: 0.24 }}
        key={key}
        chooseCard={chooseCardMock}
        compareCards={compareCardsMock}
        removeCard={removeCardMock}
      />
    )
    expect(wrapper.find('ul').hasClass('low-data')).toEqual(true)
  })

  it('should have the high-data class when its stat is above 0.5', () => {
    wrapper = shallow(
      <Card
        location="COLORADO"
        stats={{ 2004: 0.6 }}
        key={key}
        chooseCard={chooseCardMock}
        compareCards={compareCardsMock}
        removeCard={removeCardMock}
      />
    )
    expect(wrapper.find('ul').hasClass('high-data')).toEqual(true)
  })
})
