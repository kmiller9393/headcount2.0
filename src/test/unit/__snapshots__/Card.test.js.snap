// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    chooseCard={[Function]}
    compareCards={[Function]}
    compareDistricts={Array []}
    location="COLORADO"
    removeCard={[Function]}
    selected={true}
    stats={
        Object {
            "2004": 0.24,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          COLORADO
</h2>,
        Array [
          <article>
            <ul
                        className="low-data"
            >
                        2004
                        : 
                        0.24
            </ul>
</article>,
        ],
      ],
      "className": "Card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="low-data"
>
            2004
            : 
            0.24
</ul>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2004",
              ": ",
              0.24,
            ],
            "className": "low-data",
          },
          "ref": null,
          "rendered": Array [
            "2004",
            ": ",
            0.24,
          ],
          "type": "ul",
        },
        "type": "article",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            COLORADO
</h2>,
          Array [
            <article>
              <ul
                            className="low-data"
              >
                            2004
                            : 
                            0.24
              </ul>
</article>,
          ],
        ],
        "className": "Card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="low-data"
>
              2004
              : 
              0.24
</ul>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2004",
                ": ",
                0.24,
              ],
              "className": "low-data",
            },
            "ref": null,
            "rendered": Array [
              "2004",
              ": ",
              0.24,
            ],
            "type": "ul",
          },
          "type": "article",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
